### PR TITLE
guy/AddWidgetTab

### PR DESCRIPTION
**Pull request by guyyoo**

### CODE REVIEW

File: AppBuilder/platform/views/ABViewTab.js [hunk 1]
```diff
@@ -1,403 +1,73 @@
...
-            e.preventDefault();
-            return false;
-         },
-      };
-
-      return {
-         ui: _ui,
-         init: _init,
-         logic: _logic,
+class TabPopup extends EventEmitter {
```
Comment (nh758) [inline AppBuilder/platform/views/ABViewTab.js:10]: shouldn't this move to the designer? we won't use it anywhere else, right?

File: AppBuilder/platform/views/ABViewTab.js [hunk 2]
```diff
@@ -473,624 +133,1176 @@ module.exports = class ABViewTab extends ABViewTabCore {
...
-      view.settings.darkTheme = $$(ids.darkTheme).getValue();
-      view.settings.sidebarWidth = $$(ids.sidebarWidth).getValue();
-      view.settings.sidebarPos = $$(ids.sidebarPos).getValue();
-      view.settings.iconOnTop = $$(ids.iconOnTop).getValue();
+      return editedTab.save();
    }
+}
+
+module.exports = class ABViewTab extends ABViewTabCore {
+   // constructor(values, application, parent, defaultValues) {
```
Comment (nh758) [inline AppBuilder/platform/views/ABViewTab.js:229]: can wee delete instead of comment out? any reason to keep it here?

**Review by nh758 — CHANGES_REQUESTED** [review]

File: AppBuilder/platform/views/viewComponent/ABViewTabComponent.js [hunk 3]
```diff
@@ -0,0 +1,579 @@
...
+               // create a menu item from the expand option to use later
+               const expandMenu = {
+                  id: ids.expandMenu,
+                  value: L("Expand Menu"),
+                  icon: "chevron-circle-right",
+                  hidden: true,
+               };
+
+               // find out what the first option is so we can set it later
+               let selectedItem = baseView._viewComponents[0].view.id + "_menu";
```
Comment (wongpratan) [inline AppBuilder/platform/views/viewComponent/ABViewTabComponent.js:103]: ES6 string format

File: AppBuilder/platform/views/viewComponent/ABViewTabComponent.js [hunk 4]
```diff
@@ -0,0 +1,579 @@
...
+      webix.type(webix.ui.tree, {
+         baseType: "sideBar", // inherit everything else from sidebar type
+         name: "customIcons",
+         icon: (obj, common) => {
+            if (obj.icon.length)
+               return (
+                  "<span class='webix_icon webix_sidebar_icon fa fa-fw fa-" +
+                  obj.icon +
+                  "'></span>"
+               );
```
Comment (wongpratan) [inline AppBuilder/platform/views/viewComponent/ABViewTabComponent.js:58]: ES6 string format

File: AppBuilder/platform/views/viewComponent/ABViewTabComponent.js [hunk 5]
```diff
@@ -0,0 +1,579 @@
...
+            return (
+               "<span class='webix_icon webix_sidebar_icon sidebarCustomIcon'>" +
+               obj.value.charAt(0).toUpperCase() +
+               "</span>"
```
Comment (wongpratan) [inline AppBuilder/platform/views/viewComponent/ABViewTabComponent.js:62]: *_*

**Review by wongpratan — APPROVED** [review]

**Review by nh758 — APPROVED** [review]

**Pull request merged**